src/libsyntax/print/pp.rs

```diff
@@ -265,8 +265,8 @@ pub fn mk_printer<'a>(out: Box<io::Write+'a>, linewidth: usize) -> Printer<'a> {
 /// In this implementation (following the paper, again) the SCAN process is
 /// the method called 'pretty_print', and the 'PRINT' process is the method
 /// called 'print'.
-pub struct Printer<'a> {
-    pub out: Box<io::Write+'a>,
+pub struct Printer<W> {
+    pub out: W,
```

Comment:
I was trying to make it easy to get to the underlying `Vec` so I could get rid of the reference to `TraitObject` and the transmute, but I can try a different approach.